Users/services/usersService.js

```diff
@@ -0,0 +1,58 @@
+import { usersDao } from '../models';
+import bcrypt from 'bcrypt';
+import jwt from 'jsonwebtoken';
+const TOKEN_KEY = '' + process.env.SECRET_KEY;
+
+const userSignUp = async (
+  email,
+  name,
+  password,
+  account,
+  nickname,
+  phoneNumber
+) => {
+  const saltRounds = 10;
+  const salt = await bcrypt.genSalt(saltRounds);
+  const hashedPassword = await bcrypt.hash(password, salt);
+
+  const existingEmail = await usersDao.getEmail(email);
+
+  if (existingEmail.length) {
+    const error = new Error('error');
+    error.statusCode = 409;
+    throw error;
+  }
+
```

Comment:
```suggestion
) => {
  const existingEmail = await usersDao.getEmail(email);
  if (existingEmail.length) {
    const error = new Error('error');
    error.statusCode = 409;
    throw error;
  }

  const saltRounds = 10;
  const salt = await bcrypt.genSalt(saltRounds);
  const hashedPassword = await bcrypt.hash(password, salt);

```
지현님, 코드 순서를 변경하셔서 이메일이 존재하는지를 먼저 체크하면 좋을 것 같습니다 👍 
제가 왜 이 방법을 추천하는 것일까요?

Reply:
음, 소헌님께서 남겨주신 이 리뷰에 대해 생각을 조금 해보았는데요. 아직 확실하게 생각이 잡힌 것은 아니지만, 제 나름의 추측을 해보자면
이전 코드는 순서상 모순이 있다고 봅니다. 이미 존재하는 유저인지 확인하는 과정을 거치기도 전에 패스워드를 해싱하는 것은 `signup` 의 일련의 과정에 있어서 모순되는 부분이라고 생각합니다. 우선 먼저 존재하는 유저가 있는 지 확인한 뒤에, 유저가 있다면 에러 코드를 띄워주고 유저가 없다면 회원 가입을 진행시키고, 회원 가입시 입력한 유저의 패스워드를 해싱하는 것이 올바른 순서라고 생각됩니다!! 

저의 답변이 맞는 것인지, 만약 틀렸다면 왜 순서를 바꿔야 하는 것인지에 대해 설명해주시면 감사하겠습니다 소헌님! 🙇‍♀️

Reply:
수정 완료입니다! da9bdf60cc313ed0267f2af68e3ac35f7eaf44c7

Reply:
@hxyxneee 지현님 맞습니다 :) 유저가 먼저 있고 나서, 비밀번호를 암호화하는 hash 함수를 호출하는 것이 함수 호출하는 과정도 줄이고, 로직상 흐름도 자연스럽습니닷 :)